src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -1,81 +1,203 @@
+"""
+   Copyright 2017 Samuel Gulliksson
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+
 import functools
 
 import logging
 
 import flask
+from flask import current_app
 from flask.helpers import url_for
-from oic import rndstr
 from oic.oic import Client
-from oic.oic.message import ProviderConfigurationResponse, RegistrationRequest, AuthorizationResponse, EndSessionRequest
+from oic.oic.message import AuthorizationResponse
+from oic.oic.message import EndSessionRequest
+from oic.oic.message import ProviderConfigurationResponse
+from oic.oic.message import RegistrationRequest
+from oic import rndstr
 from oic.utils.authn.client import CLIENT_AUTHN_METHOD
+
+import time
 from werkzeug.utils import redirect
 
+
 logger = logging.getLogger(__name__)
 
 
+class Session(object):
+
+    """Session object for refresh tokens.
+
+    First class object for comparison of date times during session handling.
```

Comment:
Could this comment be written without reference to how it's used in the decorator below?
Draft suggestion (which needs some work :)): "Wraps comparison of date time necessary for proper session handling."